analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs

```diff
@@ -57,9 +57,18 @@ internal sealed class FieldReference : SimpleProcessor<IFieldReferenceOperationW
 
     protected override ProgramState Process(SymbolicContext context, IFieldReferenceOperationWrapper fieldReference)
     {
-        var state = fieldReference.WrappedOperation.TrackedSymbol() is { } fieldSymbol && context.State[fieldSymbol] is { } value
-            ? context.State.SetOperationValue(context.Operation, value)
-            : context.State;
+        var state = context.State;
+        if (fieldReference.WrappedOperation.TrackedSymbol() is IFieldSymbol fieldSymbol)
+        {
+            if (context.State[fieldSymbol] is { } value)
```

Comment:
For consistency - when extracted, only the extracted state should be used. In case we add some other `if` before in the future.
```suggestion
            if (state[fieldSymbol] is { } value)
```